common/primitives/src/cid.rs

```diff
@@ -0,0 +1,20 @@
+use cid::Cid;
+use sp_io::hashing::sha2_256;
+use sp_runtime::Vec;
+
+/// Multihash type for wrapping digests (support up to 64-byte digests)
+pub type Multihash = cid::multihash::Multihash<64>;
+
+/// SHA2-256 multihash code
+const SHA2_256: u64 = 0x12;
+
+/// Raw codec for CIDv1 (0x55)
+const RAW: u64 = 0x55;
+
+/// Computes a CIDv1 (RAW + SHA2-256 multihash)
+pub fn compute_cid_v1(bytes: &[u8]) -> Option<Vec<u8>> {
```

Comment:
nit: I see that you have some tests for this in the pallet but I think it would be great if you can add a test for a well known cid or something so we can be sure this works as expected

Reply:
added a deterministic test for `compute_cid_v1`